### ISSUE TITLE
replace-megaparsec-1.4.5.0 testsuite test-textStub crashed ghc-9.4.3

### ISSUE THREAD

**Issue by juhp** [jamesdbrock/replace-megaparsec#38]
This may be a ghc bug:

```
    test-textStub: internal error: Oops!  Entered absent arg Arg: $dOrd                                                             
    Type: Ord e                                                                                                                     
    In module `Replace.Megaparsec'                                                                                                  
        (GHC version 9.4.3 for x86_64_unknown_linux)
        Please report this as a GHC bug:  https://www.haskell.org/ghc/reportabug
```

**Comment by jamesdbrock** [comment]
Thanks for the bug report! Will investigate.

**Comment by jamesdbrock** [comment]
GHC v9.4.3 is the first version of GHC to include __text v2.0.1__. https://downloads.haskell.org/~ghc/9.4.3/docs/users_guide/9.4.3-notes.html

**Comment by jamesdbrock** [comment]
I upgraded GHC from v9.4.3 to v9.4.4 and the error went away and the whole test suite passed. So I guess it was some GHC problem which was fixed in v9.4.4.

Maybe this one https://gitlab.haskell.org/ghc/ghc/-/issues/22475 from the change log https://downloads.haskell.org/~ghc/9.4.4/docs/users_guide/9.4.4-notes.html
